drivers/soundwire/generic_bandwidth_allocation.c

```diff
@@ -370,16 +402,89 @@ static int sdw_compute_bus_params(struct sdw_bus *bus)
 	}
 
 	if (i == clk_values) {
-		dev_err(bus->dev, "%s: could not find clock value for bandwidth %d\n",
+multilane:
+		dev_dbg(bus->dev, "%s: could not find clock value for bandwidth %d, checking multi-lane\n",
 			__func__, bus->params.bandwidth);
-		return -EINVAL;
+
+		m_rt = list_last_entry(&bus->m_rt_list, struct sdw_master_runtime, bus_node);
+		s_rt = list_first_entry(&m_rt->slave_rt_list, struct sdw_slave_runtime, m_rt_node);
+		slave_prop = &s_rt->slave->prop;
+
+		/*
+		 * Find an available manager lane for the first Peripheral
+		 * Testing the first Peripheral is enough because we can't use multi-lane
+		 * if we can't find any available lane for the first Peripheral
+		 */
+		for (i = 1; i < SDW_MAX_LANES; i++) {
+			if (!slave_prop->lane_maps[i])
+				continue;
+
+			required_bandwidth = 0;
+			list_for_each_entry(m_p_rt, &m_rt->port_list, port_node) {
+				required_bandwidth += m_rt->stream->params.rate *
+					hweight32(m_p_rt->ch_mask) *
+					m_rt->stream->params.bps;
+			}
+			if (required_bandwidth <= curr_dr_freq - bus->lane_used_bandwidth[i]) {
+				/* Check if m_lane is connected to all Peripherals */
+				if (!is_all_peripherals_connected(m_rt, slave_prop->lane_maps[i])) {
+					dev_dbg(bus->dev,
+						"some Peripherals are not connected to %d\n",
+						slave_prop->lane_maps[i]);
+					continue;
```

Comment:
so here you continue to the next lane, but you've already increased required_bandwidth, so you will add it a second time.

Reply:
> so here you continue to the next lane, but you've already increased required_bandwidth, so you will add it a second time.

No, `required_bandwidth` is set to 0 at the beginning of the loop. So when it continue, `required_bandwidth = 0;` will be called first, then increase `required_bandwidth`.